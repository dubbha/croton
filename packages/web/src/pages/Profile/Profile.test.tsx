import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

jest.mock('components/Header', () => {
  const React = require('react');
  return {
    Header: () => <div>Header</div>,
  };
});

jest.mock('components/Footer', () => {
  const React = require('react');
  return {
    Footer: () => <div>Footer</div>,
  };
});

jest.mock('components/ProfileHeader', () => {
  const React = require('react');
  return {
    ProfileHeader: () => <div>ProfileHeader</div>,
  };
});
jest.mock('components/ManageSocials/FbAuth', () => {
  const React = require('react');
  return {
    FbAuth: () => <button />,
  };
});
jest.mock('components/ManageSocials/GoogleAuth', () => {
  const React = require('react');
  return {
    GoogleAuth: () => <button />,
  };
});

jest.mock('connected-react-router', () => ({
  push: (path: string) => ({ type: 'callHistoryMethod', payload: { path } }),
}));

describe('pages/Profile', () => {
  it('should render successfully if user is logged in', () => {
    jest.isolateModules(() => {
      const fn = jest.fn();
      jest.doMock('react-redux', () => ({
        useDispatch: () => fn,
        useSelector: () => ({
          token: 'TOKEN',
          firstName: 'FIRST_NAME',
          lastName: 'LAST_NAME',
          email: 'EMAIL',
        }),
      }));

      const { Profile } = require('./Profile');

      const { container } = render(<Profile match={{ path: '/profile' }} />, {
        wrapper: MemoryRouter,
      });

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should redirect to sing in if user is not logged in', () => {
    jest.isolateModules(() => {
      const fn = jest.fn();
      jest.doMock('react-redux', () => ({
        useDispatch: () => fn,
        useSelector: () => ({
          token: null,
          firstName: null,
          lastName: null,
          email: null,
        }),
      }));

      const { Profile } = require('./Profile');

      render(<Profile match={{ path: '/profile' }} />, {
        wrapper: MemoryRouter,
      });

      expect(fn).toBeCalledWith({
        type: 'callHistoryMethod',
        payload: { path: '/signin' },
      });
    });
  });
});
