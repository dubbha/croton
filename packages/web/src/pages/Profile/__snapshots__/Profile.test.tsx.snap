// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Profile should render successfully if user is logged in 1`] = `
<div
  class="container"
>
  <div>
    Header
  </div>
  <div>
    FIRST_NAME LAST_NAME [EMAIL]
  </div>
  <div>
    Footer
  </div>
</div>
`;
